docs/README.md

```diff
@@ -1,9 +1,27 @@
 ## Technical Overview
 
-Kubernetes allows for extensions to its architecture in the form of custom
-resources: <https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/>.
-KubeVirt represents virtual machines as custom resources and manages changes
-to libvirt domains based on the state of those resources.
+Kubernetes allows for extensions to its architecture via
+[*custom resources*]( https://Kubernetes.io/docs/concepts/extend-Kubernetes/api-extension/custom-resources/),
+which add a new endpoint in the Kubernetes API that stores and retrieves a
+collection API objects of a certain kind. However, the *custom resources*
```

Comment:
I find the "However" sentence mostly repetitive. I think you should just explain why controllers are needed, not why CRDs are not enough.

Reply:
Makes sense, let me fix that!